src/resty/gcp/request/credentials/accesstoken.lua

```
@@ -104,7 +120,22 @@ local function GetAccessTokenByWI()
 end
 
 local AccessToken = {}
-AccessToken.__index = AccessToken
+
+function AccessToken.__index(self, key)
+    if key == "token" then
+        if self:needsRefresh() then
+            local ok, err = self:refresh()
+            if not ok then
+                ngx.log(ngx.ERR, "[accesstoken] auto refresh failed: ", tostring(err))
+                return nil
```

This now makes a silent fail and access token will be nil if refresh failed

Let's add proper handling here https://github.com/Kong/lua-resty-gcp/blob/301e8dd228f868af8a65881a9c79b125c9add38b/src/resty/gcp/init.lua#L77